env0/data_aws_credentials.go

```diff
@@ -82,7 +82,7 @@ func getAwsCredentialsByName(name interface{}, meta interface{}) (client.ApiKey,
 func getAwsCredentialsById(id string, meta interface{}) (client.ApiKey, diag.Diagnostics) {
 	apiClient := meta.(client.ApiClientInterface)
 	credentials, err := apiClient.AwsCredentials(id)
-	if credentials.Type != "aws_assumed_role" {
+	if credentials.Type != "AWS_ASSUMED_ROLE_FOR_DEPLOYMENT" {
```

Comment:
please DRY the value `AWS_ASSUMED_ROLE_FOR_DEPLOYMENT`
DRY - don't repeat yourself

which means, export the value to a shared constant so both places can use it